otelcollector/configuration-reader-builder/main.go

```diff
@@ -209,6 +209,7 @@ func taHealthHandler(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(status)
 		fmt.Fprintln(w, message)
 	}
+	defer resp.Body.Close()
```

Comment:
@rashmichandrashekar Thanks for fixing this. Can you also add a condition like this [here](https://github.com/Azure/prometheus-collector/blob/main/otelcollector/shared/process_utilities.go#L418) ?

Reply:
Thanks @bragi92. Looking at this code I am wondering, this http request is not closed as well, which might lead to mem leaks. Did you see that behavior?

Reply:
No, I haven't seen memory leaks yet for this. This has been running on our ci-prod cluster for 11 days now but surpsingly no extra memory usage ~ maybe its a very small leak? I think it should be closed either way though so that it doesn't have a chance of misbehaving.

Reply:
Yes, It might be very small depending on the size of the response. I think we need to look at all the http calls we are making and make sure we are closing them. Lets sync during scrum, we can look at memory patterns too.

Reply:
So, from the code looks like the liveness check for healthz endpoint might not cause memory leak because it is only called a few times and then we give up. So, it doesnt add up over time. But I added a close for this anyway and dont see any other http calls other than these.